keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,331 @@
+
+### How to use WorkloadPriorityCLass on MPIJob
+
+The `workloadPriorityClass` is specified through a label `kueue.x-k8s.io/workload-priority-class`.
+This is same as other CRDs like `RayJob`.
+
+```yaml
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: pi
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/workload-priority-class: sample-priority
+spec:
+.....
+```
+
+### How workloads are created from Jobs
+
+There are three scenarios for creating a workload from a job.
+The same applies to CRDs other than `Job` (such as `RayJob`).
+
+1. A job specifies both `workload's priority` and `pod's priority`
+2. A job specifies only `workload's priority`
+3. A job specifies only `pod's priority`
+
+#### 1. A job specifies both `workload's priority` and `pod's priority`
+
+When creating this yaml, the `workloadPriorityClass` sample-priority is used for the `workload's priority`.
+On the other hand, the `priorityClass` high-priority is used for the `pod's priority`.
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/workload-priority-class: sample-priority
+spec:
+  priorityClassName: high-priority
+  parallelism: 3
+  completions: 3
+  suspend: true
+  template:
+    spec:
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+      restartPolicy: Never
+```
+
+#### 2. A job specifies only `workload's priority`
+
+When creating this yaml, the `workloadPriorityClass` sample-priority is used for the `workload's priority`.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/workload-priority-class: sample-priority
+spec:
+  parallelism: 3
+  completions: 3
+  suspend: true
+  template:
+    spec:
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+      restartPolicy: Never
+```
+
+#### 3. A job specifies only `pod's priority`
+
+When creating this yaml, the `PriorityClass` high-priority is used for the `workload's priority`.
+This is basically same as current implementation of workload.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  priorityClassName: high-priority
+  parallelism: 3
+  completions: 3
+  suspend: true
+  template:
+    spec:
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+      restartPolicy: Never
+```
+
+### How to expand Priority utility
+
+When referencing the `priorityClass` in the [Priority utility](https://github.com/kubernetes-sigs/kueue/blob/ba404ad282c35cf1d6b15d07643935fffbcc1835/pkg/util/priority/priority.go) function, `workloadPriorityClass` is checked first.
```

Comment:
We should have different functions. If the user sets the priority via annotation, we should search for the WorkloadPriorityClass.
If the user only sets pod priority, we look for the k8s PriorityClass.